components/invitation_modal/add_to_channels.tsx

```diff
@@ -78,6 +82,12 @@ export default function AddToChannels(props: Props) {
         placeholderChannelName = props.currentChannel.display_name;
     }
 
+    // don't show the add to channels if it is not a free trial nor a paid subscription
+    // in case the enduser overpasses restrictions by DOM manipulation (see MM-47228)
+    if (props.isCloud && !props.isCloudFreeTrial && !props.isPaidSubscription) {
+        return null;
+    }
```

Comment:
Should we be checking for license here instead? This would only apply for cloud but the feature is also available in self-managed

Reply:
I think this only affects cloud, afaik self-hosted only allow or display something related to guests once you are on trial or have a valid license, otherwise the guest access is disabled everywhere